### ISSUE TITLE
Remaining and Duration columns semi-broken in Students tab

### ISSUE THREAD

**Issue by echuber2** [PrairieLearn/PrairieLearn#2149]
If you view a given `pl/course_instance/nnnnn/instructor/assessment/nnnnnnn/instances` page (the "Students" tab for a given assessment instance), the Remaining and Duration columns don't always seem to display correct info:

The Duration column appears to show "0s" as the duration for a session that closed due to time limit running out. Example available here (may not be persistent if we need to correct the grade manually later and that changes something): https://prairielearn.engr.illinois.edu/pl/course_instance/64437/instructor/assessment/1413729/instances Possibly connected to the recent PR about LTI https://github.com/PrairieLearn/PrairieLearn/pull/2144 This was a situation where the student interacted with the first variant for about 15 seconds, and then may have had some issue that led to logging in again through the direct PrairieLearn URL that should not be available, so there may be more than one thing going on. In this case, it's also strange that the session timed out after 100 minutes when it should have timed out after 90 minutes. (Maybe this warrants a separate issue. I'm testing how long it takes for another session to time out, now.)

The Remaining column seems to show "0 min" while a live exam is in session. (Tested this several times myself over the past few months, but can't link an example that will persist.)

**Comment by echuber2** [comment]
With regard to the Duration issue, it appears that the column is only updated upon the student executing an action, not on-demand upon "Students" tab refresh. I had another testing session open and saw the Duration jump from "15s" to "12m" only after the dummy test taker account executed an exam action, e.g. save the open variant.

On a separate note, it's inconsistent that the Duration column uses a `-m` suffix while the Remaining column uses a `-(space)min` suffix.

**Comment by echuber2** [comment]
> In this case, it's also strange that the session timed out after 100 minutes when it should have timed out after 90 minutes. (Maybe this warrants a separate issue. I'm testing how long it takes for another session to time out, now.)

The session I started and allowed to expire (launched in incognito tab session and just closed the window instead of finalizing the exam or logging out) apparently finally auto-closed after _95_ minutes this time. It's strange because the countdown timer shown to a student actively taking the exam seems to keep time properly. The "Duration" column after closure shows 16m, so I guess "Duration" does only reflect the span between user interactions, not time from start to close.

**Comment by tdy** [comment]
@mwest1066 Could you clarify what the "Durations" column is meant to convey?

**Comment by echuber2** [comment]
> @mwest1066 Could you clarify what the "Durations" column is meant to convey?

Bump...

**Comment by jonatanschroeder** [comment]
> @mwest1066 Could you clarify what the "Durations" column is meant to convey?

@mwest1066 I can't see any formal description for how the Duration can be explained to an instructor. I have a vague idea of its meaning, but I'd appreciate it if you could propose a more complete description to add as a helper to this column in #4560.

**Comment by mwest1066** [comment]
> @mwest1066 I can't see any formal description for how the Duration can be explained to an instructor. I have a vague idea of its meaning, but I'd appreciate it if you could propose a more complete description to add as a helper to this column in #4560.

> The "Duration" is the amount of time that a student has spent actively working on the assessment. The duration time measurement begins when the student starts the assessment and continues until the most recent answer submission. A student is considered to be actively working if they have at least one answer submission per hour, so the duration measurement is paused if there is a gap of more than one hour between answer submissions.

As an example:
```
08:00 start assessment
08:30 submit answer
09:00 submit answer
----- gap of more than one hour
11:00 submit answer
11:30 submit answer
12:00 submit answer
```

In the above example, the "duration" would be 2 hours. This is 1 hour from 8 to 9, and another hour from 11 to 12. The two-hour gap from 9 to 11 is not counted as part of the duration.

**Comment by jonatanschroeder** [comment]
Do other events (e.g., open instance page) also count in the duration?

**Comment by mwest1066** [comment]
> Do other events (e.g., open instance page) also count in the duration?

Currently, no.
